build: Fix "make create-github-release"

camscale opened the pull request:
Fix the `create-github-release` make target to use "go run" to run the
tool that creates the github release. This used to run a tool in this
repository that was pre-built, so the "go run" was not needed before.
Now the tool has moved to a separate repository, this is required.

camscale committed:
build: Fix "make create-github-release"

Fix the `create-github-release` make target to use "go run" to run the
tool that creates the github release. This used to run a tool in this
repository that was pre-built, so the "go run" was not needed before.
Now the tool has moved to a separate repository, this is required.

zmb3 commented on Makefile:
```diff
@@ -1826,7 +1826,7 @@ changelog:
 # For more information on release notes generation see:
 #   https://github.com/gravitational/shared-workflows/tree/gus/release-notes/tools/release-notes#readme
-RELEASE_NOTES_GEN = github.com/gravitational/shared-workflows/tools/release-notes@latest
+RELEASE_NOTES_GEN = go run github.com/gravitational/shared-workflows/tools/release-notes@latest
```
Do we need a variable for this when it's only used in one place?

IMO the extra indirection just makes it harder to piece together the full command.

camscale replied:
It's pretty common with make to use variables to define commands so that you can override them as needed (`make CC=gcc` for example), but we probably don't really need that here. Updated. And I updated the `changelog` target too for which your comment also applies.

camscale committed:
build: Remove single-use vars for simplicity

Remove the `CHANGELOG` and `RELEASE_NOTES_GEN` vars that contained the
name of a Go main package to run in favor of a straightforward "go run"
command.

fheinecke reviewed (approved):
We now have release tooling for the changelog, and I can add it for the release notes tool with about 5 minutes of work. Do you have an opinion on either:
* Changing `go run` to `docker run` (removing the need to compile and install)
* Requiring that the tool is pre-installed (like pretty much all other commands), maybe adding a "bootstrap" target
